### ISSUE TITLE
Has any able to train successfully?

### ISSUE THREAD

**Issue by soon-yau** [LambdaLabsML/examples#30]
I wonder if anyone has successfully finetune the model? I have difficulty training using the code, either finetuning from StableDiffusion checkpoint or from scratch.

I noticed that after around 2000 steps, the loss spiked from 0.1 to 1 then stuck there. Inference from checkpoints show that when loss was 0.1, the generated images was "ok" i.e. if finetune, still able to generate recognizable shape of my custom dataset despite not pretty; if from scratch, can see blob of noise. However, when the loss stuck at 1.0, all it can generate is black.

I tried different random seed but didn't help.

**Comment by gstswwx** [comment]
try a smaller learning rate

**Comment by KyonP** [comment]
I am also struggling to finetune with a custom dataset on the cartoon domain. 

Following suggestions, I have tried smaller batch size (1 with accumulated batch size 4) and learning rates, and outputs look like partially illustrated characters appeared all over an image.

Still, I couldn't achieve successful results.

If there is other advice, I would be very grateful.

**Comment by soon-yau** [comment]
I have tried smaller learning rate but I think the instability comes from small batch size of 1, which is limited by my GPU memory. 

I then switched to official Stable Diffusion repo (basically a LDM), reduce the Unet channel dimension, set batch size=16 and train the mode from scratch, then it is working.

**Comment by KyonP** [comment]
> I have tried smaller learning rate but I think the instability comes from small batch size of 1, which is limited by my GPU memory.
> 
> I then switched to official Stable Diffusion repo (basically a LDM), reduce the Unet channel dimension, set batch size=16 and train the mode from scratch, then it is working.

I'll try this thanks,

BTW, is this repo (LambdaLabs Poke'mon examples) significantly different from the official Stable Diffusion?

I am just asking because of curiosity since I haven't dug into this repo enough.

**Comment by KyonP** [comment]
I only get these noisy images. 

![image](https://user-images.githubusercontent.com/6139342/202453006-d56cb87c-e525-42ce-befc-80fb69249484.png)

I am not sure where did I make mistake. any suggestions would be greatly appreciated.

**Comment by BannedUseR-X** [comment]
@KyonP Maybe you forgot add '--' before 'finetune from ...' in command.

**Comment by nocol0101001** [comment]
> @KyonP Maybe you forgot add '--' before 'finetune from ...' in command.

!(python main.py
-t
--base configs/stable-diffusion/pokemon.yaml
--gpus "0,"
--scale_lr False
--num_nodes 1
--check_val_every_n_epoch 10
--finetune_from "$ckpt_path"
data.params.batch_size="2"
lightning.trainer.accumulate_grad_batches="1"
data.params.validation.params.n_gpus="$NUM_GPUS"
)

Hello, what I get from training like this is still a mass of noise, what is the cause. Thanks

**Comment by BannedUseR-X** [comment]
> 

try running txt2img.py, using the checkpoint in your $ckpt_path. If you still get noisy images, maybe your checkpoint model is incorrect.

**Comment by BannedUseR-X** [comment]
> > @KyonP Maybe you forgot add '--' before 'finetune from ...' in command.
> 
> !(python main.py -t --base configs/stable-diffusion/pokemon.yaml --gpus "0," --scale_lr False --num_nodes 1 --check_val_every_n_epoch 10 --finetune_from "$ckpt_path" data.params.batch_size="2" lightning.trainer.accumulate_grad_batches="1" data.params.validation.params.n_gpus="$NUM_GPUS" )
> 
> Hello, what I get from training like this is still a mass of noise, what is the cause. Thanks

try running txt2img.py, using the checkpoint in your $ckpt_path. If you still get noisy images, maybe your checkpoint model is incorrect.

**Comment by nocol0101001** [comment]
Hello, this is my training and testing code, and last.ckpt is the checkpoint I saved after fine-tuning.The checkpoint is about 13 G in size. Is there anything wrong? Greatful




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "LambdaLabsML/examples"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年5月4日(星期四) 下午5:36
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [LambdaLabsML/examples] Has any able to train successfully? (Issue #30)





   
@KyonP Maybe you forgot add '--' before 'finetune from ...' in command.
  
!(python main.py -t --base configs/stable-diffusion/pokemon.yaml --gpus "0," --scale_lr False --num_nodes 1 --check_val_every_n_epoch 10 --finetune_from "$ckpt_path" data.params.batch_size="2" lightning.trainer.accumulate_grad_batches="1" data.params.validation.params.n_gpus="$NUM_GPUS" )
 
Hello, what I get from training like this is still a mass of noise, what is the cause. Thanks
  
try running txt2img.py, using the checkpoint in your $ckpt_path. If you still get noisy images, maybe your checkpoint model is incorrect.

**Comment by BannedUseR-X** [comment]
> > > > @KyonP Maybe you forgot add '--' before 'finetune from ...' in command.
> > > 
> > > 
> > > !(python main.py -t --base configs/stable-diffusion/pokemon.yaml --gpus "0," --scale_lr False --num_nodes 1 --check_val_every_n_epoch 10 --finetune_from "$ckpt_path" data.params.batch_size="2" lightning.trainer.accumulate_grad_batches="1" data.params.validation.params.n_gpus="$NUM_GPUS" )
> > > Hello, what I get from training like this is still a mass of noise, what is the cause. Thanks
> > 
> > 
> > try running txt2img.py, using the checkpoint in your $ckpt_path. If you still get noisy images, maybe your checkpoint model is incorrect.
> > Hello, this is my training and testing code, and last.ckpt is the checkpoint I saved after fine-tuning.The checkpoint is about 13 G in size. Is there anything wrong? Greatful
> > ![image](https://user-images.githubusercontent.com/102145044/236169860-d8e3b999-97e0-4842-8e87-5c51e68ea0ab.png)
> > ![image](https://user-images.githubusercontent.com/102145044/236169981-2c810963-6dd3-42d8-91dc-5a38b937043c.png)

Still noisy pics? It looks all right. Check the fine-tuning logdir, did you get correct pics? You can send me more information by email -> wangsheng_0205@163.com.

**Comment by lvsi-qi** [comment]
> > > > > 也许您忘记在“微调来自...”之前添加“--”在命令中。
> > > > 
> > > > 
> > > > !（python main.py --base configs/stable-diffusion/pokemon.yaml --gpus “0，” --scale_lr false --num_nodes 1 --check_val_every_n_epoch 10 --finetune_from “$ckpt_path” data.params.batch_size=“2” lightning.trainer.accumulate_grad_batches=“1” data.params.validation.params.n_gpus=“$NUM_GPUS” ）你好，我从这样的训练中得到的仍然是一团噪音，是什么原因。谢谢
> > > 
> > > 
> > > 尝试使用 $ckpt_path 中的检查点运行 txt2img.py。如果您仍然收到噪点图像，则可能是您的检查点模型不正确。您好，这是我的训练和测试代码，last.ckpt 是我在微调后保存的检查点。检查点的大小约为 13 G。有什么问题吗？太好了
> > > ![image](https://user-images.githubusercontent.com/102145044/236169860-d8e3b999-97e0-4842-8e87-5c51e68ea0ab.png)
> > > ![image](https://user-images.githubusercontent.com/102145044/236169981-2c810963-6dd3-42d8-91dc-5a38b937043c.png)
> 
> 仍然嘈杂的照片？看起来没问题。检查微调日志，你得到正确的照片吗？您可以通过电子邮件向我发送更多信息 -> [wangsheng_0205@163.com](mailto:wangsheng_0205@163.com)。

Hello, I would like to ask how much epoch is and how often does checkpoint save it?

**Comment by JaosonMa** [comment]
https://github.com/LambdaLabsML/examples/blob/main/stable-diffusion-finetuning/pokemon_finetune.ipynb
 i use this to retrain pokemon:
`export MODEL_NAME="runwayml/stable-diffusion-v1-5"
export OUTPUT_DIR="./pokemon"
#export HUB_MODEL_ID="pokemon-lora"
#export DATASET_NAME="lambdalabs/pokemon-blip-captions"
export TRAIN_DIR="./images/train"
export CACHE_DIR="./text_to_image/cache_dir"


export CUDA_VISIBLE_DEVICES=1

accelerate launch --mixed_precision="fp16"  train_text_to_image_lora.py \
  --pretrained_model_name_or_path=$MODEL_NAME \
  --train_data_dir=${TRAIN_DIR}   \
  --cache_dir=${CACHE_DIR} \
  --dataloader_num_workers=8 \
  --resolution=512 --center_crop --random_flip \
  --train_batch_size=1 \
  --gradient_accumulation_steps=4 \
  --max_train_steps=15000 \
  --learning_rate=1e-04 \
  --max_grad_norm=1 \
  --lr_scheduler="cosine" --lr_warmup_steps=0 \
  --output_dir=${OUTPUT_DIR} \
  --checkpointing_steps=1000 \
  --report_to=wandb \
  --resume_from_checkpoint="latest" \
  --validation_prompt="A pokemon with blue eyes." \
  --seed=1337`
  but the loss continue jump Oscillating around 0.5. why? i also try use batch_size=4,same result
![image](https://github.com/LambdaLabsML/examples/assets/23551774/b74ebcee-3c59-4b7e-a73b-9d91a5e8f205)
